pkg/cmd/registry/rule/list/list.go

```diff
@@ -71,17 +72,27 @@ func NewListCommand(f *factory.Factory) *cobra.Command {
 		Args:    cobra.NoArgs,
 		RunE: func(cmd *cobra.Command, _ []string) (err error) {
 
-			cfg, err := opts.Config.Load()
+			context, err := opts.Profiles.Load()
 			if err != nil {
 				return err
 			}
 
-			instanceID, ok := cfg.GetServiceRegistryIdOk()
-			if !ok {
-				return opts.localizer.MustLocalizeError("artifact.cmd.common.error.noServiceRegistrySelected")
+			profileHandler := &profileutil.ContextHandler{
+				Context:   context,
+				Localizer: opts.localizer,
 			}
 
-			opts.registryID = instanceID
+			conn, err := opts.Connection(connection.DefaultConfigRequireMasAuth)
+			if err != nil {
+				return err
+			}
+
+			registryInstance, err := profileHandler.GetCurrentRegistryInstance(conn.API().ServiceRegistryMgmt())
```

Comment:
Should profileHandler have embedded instance of the API object?

Reply:
Since api object is specific to the service, I don't think we should embed it.